infra/templates/az-isolated-service-single-region/ase.tf

```diff
@@ -55,10 +55,11 @@ module "app_service" {
   service_plan_name                = module.service_plan.service_plan_name
   service_plan_resource_group_name = azurerm_resource_group.admin_rg.name
   app_insights_instrumentation_key = module.app_insights.app_insights_instrumentation_key
+  vault_uri                        = module.keyvault.keyvault_uri
   azure_container_registry_name    = module.container_registry.container_registry_name
   docker_registry_server_url       = module.container_registry.container_registry_login_server
   docker_registry_server_username  = module.acr_service_principal_acrpull.service_principal_application_id
-  docker_registry_server_password  = format("@Microsoft.KeyVault(SecretUri=%s)", "module.acr_service_principal_password.keyvault_secret_ids[0]") #data.azurerm_key_vault_secret.acr_password.id)
+  docker_registry_server_password  = format("@Microsoft.KeyVault(SecretUri=%s)", module.acr_service_principal_password.keyvault_secret_ids[0]) #data.azurerm_key_vault_secret.acr_password.id)
```

Comment:
Think I found a fix for the keyvault versioning problem in this blog post:  pumpingco.de/ blog/ azure-key-vault-references-for-application-settings-with-terraform/

Reply:
The approach from the above blog failed to pan out. Actual approach can be viewed in the block of code containing the key-vault-secret module output.